crates/swc/src/config/mod.rs

```diff
@@ -365,6 +365,11 @@ impl Options {
                 syntax.decorators()
             ),
             import_assertions(),
+            // Do a resolver pass after decorators as it might
+            // emit runtime declarations and do it before
+            // type stripping as we need to know scope information
+            // for emitting enums and namespaces.
+            resolver_with_mark(top_level_mark),
```

Comment:
I feel like we should be able to improve the decorators transform to handle having a resolver pass first. Not a big deal though.